lib/index.js

```diff
@@ -1,55 +1,61 @@
+const axios = require('axios');
+const FormData = require('form-data');
 const { Readable } = require('stream');
 
+const uploadVideo = require("./uploadVideo");
+const uploadImage = require("./uploadImage");
+
 module.exports = {
-  init(config) {
+  init: (config) => {
     const accountId = config.accountId;
     const apiKey = config.apiKey;
     return {
-      upload(file) {
-        const stream = new Readable({
-          read() {
-            this.push(file.buffer);
-            this.push(null);
-          },
-        })
-        var data = new FormData();
-        data.append("file", stream, `${file.hash}${file.ext}`);
-        return axios({
-          method: "POST",
-          url: `https://api.cloudflare.com/client/v4/accounts/${accountId}/images/v1`,
-          headers: {
-            Authorization: `Bearer ${apiKey}`,
-            ...data.getHeaders(),
-          },
-          data: data,
-        }).then((response) => {
-          const result = response.data.result;
-          const filename = result.filename;
-          const split = filename.split('.');
-          const type = split.length > 0 ? split[split.length - 1] : '';
-          file.url = result.variants[0];
-          file.provider_metadata = {
-            public_id: result.id,
-            resource_type: type,
-          };
-          return file;
-        });
+      upload: async (file) => {
+        switch (file.type.split('/')[0]) {
+          case 'video':
+            const uploadVideoResponse = await uploadVideo(file, accountId, apiKey);
+            const vresult = uploadVideoResponse;
+            file.url = 'https://watch.cloudflarestream.com/'+vresult._response.headers['stream-media-id'];
+            file.provider_metadata = {
+              public_id: vresult._response.headers['stream-media-id'],
+              hls_url: 'https://videodelivery.net/'+vresult._response.headers['stream-media-id']+'/manifest/video.m3u8w',
+              dash_url: 'https://videodelivery.net/'+vresult._response.headers['stream-media-id']+'/manifest/video.mpd',
+            };
+            return file
+          case 'image':
+            const uploadImageResponse = await uploadImage(file, accountId, apiKey);
+            const result = uploadImageResponse.data.result;
+            const filename = result.filename;
+            const split = filename.split('.');
+            const type = split.length > 0 ? split[split.length - 1] : '';
+            file.url = result.variants.find(variant => variant.endsWith('/cms'));
+            file.provider_metadata = {
+              public_id: result.id,
+              resource_type: type,
+              variant_url: file.url.substr(0, file.url.lastIndexOf('/'))
+            };
+            return file
+          default:
+            throw new Error('Unhandled File Type');
+        }
       },
-      delete(file) {
-        const { public_id } = file.provider_metadata;
+      delete(delete_id, filetype) {
+        var type = '';
```

Comment:
can we move this to a separate utility and do an integration test of it separately

Reply:
moved to another utility and integration test fully functional - could be expanded to cover different file types etc